src/components/Item.jsx

```diff
@@ -1,10 +1,25 @@
 import React from 'react';
 import db from '../lib/firebase';
+import calculateEstimate from '../lib/estimates';
 
-const Item = ({ name, id, lastPurchasedDate }) => {
-  const checkHandler = () => {
+const Item = ({
+  name,
+  id,
+  frequency,
+  lastPurchasedDate,
+  numberOfPurchases,
+}) => {
+  const checkHandler = async () => {
     if (lastPurchasedDate === null || !checked) {
-      db.collection('items').doc(id).update({ lastPurchasedDate: new Date() });
+      let nextPurchaseEstimate = getPurchaseInterval();
+      await db
+        .collection('items')
+        .doc(id)
+        .update({
+          lastPurchasedDate: new Date(),
+          numberOfPurchases: numberOfPurchases + 1,
+          frequency: nextPurchaseEstimate,
```

Comment:
I think changing the `frequency` name with something more explicit like `nextPurchaseEstimate` or just `nextPurchase` could make it more self-explanatory and clear. The reason is that `frequency` value is not only **how soon** the user is going to purchase the item, but also it is an **estimate** as to when the user is likely to purchase the same item again. 
(_Keep in mind that if you change its name here in the **Item** component, you will also need to change it in other places. For example in **CreateItem** component_)

This is just a tiny suggestion. You can ignore it.

Reply:
As per your suggestion `frequency ` is now called `nextPurchase`. Thanks Luka! 🥇